test/end-to-end/src/non-existent-element/empty-cmp.tsx

```diff
@@ -0,0 +1,11 @@
+import { Component, h, Host } from '@stencil/core';
+
+@Component({
+  tag: 'empty-cmp',
+  shadow: true,
+})
+export class EmptyComponent {
+  render() {
+    return <Host>I have no children!</Host>;
+  }
+}
```

Comment:
Can we add a similar component (and matching suite of tests) where it doesn't use `shadow`? I don't think it matters if it's totally non-encapsulated or used `scoped` - I just want to make sure that in the case there's not an actual shadow root, we handle that case as well.